vault_core/src/vault.rs

```diff
@@ -845,6 +847,106 @@ impl Vault {
         })
     }
 
+    /// Checks to see if the minimum amount out is acceptable
+    /// acceptable being defined as slippage being larger or equal to the minimum slippage
+    /// as well as accounting for withdraw fees
+    pub fn check_min_supported_mint_out(
+        &self,
+        vrt_amount_in: u64,
+        min_supported_mint_out: u64,
+    ) -> Result<(), VaultError> {
+        if vrt_amount_in == 0 {
+            msg!("Amount in is zero");
+            return Err(VaultError::VaultBurnZero);
+        } else if vrt_amount_in > self.vrt_supply() {
+            msg!("Amount exceeds vault VRT supply");
+            return Err(VaultError::VaultInsufficientFunds);
+        }
+
+        let fee_amount = self.calculate_withdraw_fee(vrt_amount_in)?;
+        let amount_to_burn = vrt_amount_in
+            .checked_sub(fee_amount)
+            .ok_or(VaultError::VaultUnderflow)?;
+
+        let amount_out: u64 = (amount_to_burn as u128)
+            .checked_mul(self.tokens_deposited() as u128)
+            .and_then(|x| x.checked_div(self.vrt_supply() as u128))
+            .and_then(|x| x.try_into().ok())
+            .ok_or(VaultError::VaultOverflow)?;
+
+        let amount_out_delta = amount_out.saturating_sub(min_supported_mint_out);
+        let calculated_slippage = amount_out_delta
+            .checked_mul(MAX_BPS as u64)
+            .and_then(|x| x.checked_div(amount_out))
+            .ok_or(VaultError::VaultOverflow)?;
+
+        if calculated_slippage < Self::MIN_WITHDRAWAL_SLIPPAGE_BPS as u64 {
+            msg!(
+                "Calculated slippage {} is less that the minimum slippage {}",
+                calculated_slippage,
+                Self::MIN_WITHDRAWAL_SLIPPAGE_BPS
+            );
+            return Err(VaultError::SlippageError);
+        }
+
+        let calculated_min_amount_out =
+            self.calculate_min_supported_mint_out(vrt_amount_in, calculated_slippage as u16)?;
```

Comment:
add comment about this being a sanity check - seemed redundant to me at first

Reply:
10-4